lib/httplib/httplib.go

```diff
@@ -305,11 +305,17 @@ func OriginLocalRedirectURI(redirectURL string) (string, error) {
 		return "", trace.BadParameter("Invalid scheme: %s", parsedURL.Scheme)
 	}
 
+	// Make sure User field does not contain password to prevent basic auth
+	if parsedURL.User != nil {
+		_, hasPassword := parsedURL.User.Password()
```

Comment:
I don't think we even need the password check - it's probably enough to fail if `parsedURL.User != nil`.

We should also write a test for this change (or update an existing test if there is one already).